sdk/keyvault/keyvault-certificates/src/certificatesModels.ts

```diff
@@ -624,3 +624,51 @@ export interface RecoverDeletedCertificateOptions extends coreHttp.OperationOpti
  * An interface representing optional parameters for {@link restoreCertificateBackup}.
  */
 export interface RestoreCertificateBackupOptions extends coreHttp.OperationOptions {}
+
+/**
+ * The shape of the contact information for the vault certificates.
+ */
+export interface CertificateContactAll {
+  /**
+   * Email address.
+   */
+  emailAddress: string;
+  /**
+   * Name.
+   */
+  name: string;
+  /**
+   * Phone number.
+   */
+  phone: string;
+}
+
+/**
+ * RequireAtLeastOne allows us to potentially ignore all the other properties of a type,
+ * but also to enforce at least one.
+ */
+export type RequireAtLeastOne<T> = {
+  [K in keyof T]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<keyof T, K>>>;
+}[keyof T];
+
+/**
+ * The contact information for the vault certificates.
+ * Each contact will have just one of the properties of CertificateContactAll,
+ * which are emailAddress, name or phone.
+ */
+export type CertificateContact = RequireAtLeastOne<CertificateContactAll> | undefined;
```

Comment:
I feel like it doesn't need to be maybe undefined as its usage below is optional

```suggestion
export type CertificateContact = RequireAtLeastOne<CertificateContactAll>;
```

Reply:
I need to test this before I commit it! I thought I was getting an error when I tried it.

Reply:
It works! Committing